socksio/socks5.py

```diff
@@ -189,30 +219,28 @@ def __init__(self) -> None:
     def state(self) -> SOCKS5State:
         return self._state
 
-    def authenticate(self, methods: typing.List[SOCKS5AuthMethod]) -> None:
-        auth_request = SOCKS5AuthRequest(methods)
-        self._data_to_send += auth_request.dumps()
+    @singledispatchmethod
+    def send(self, request: SOCKS5RequestType) -> None:
+        raise NotImplementedError
```

Comment:
`raise NotImplementedError()`

Reply:
I believe you can omit the brackets but I can't find it in the docs right now, never liked it anyway 😄